nomad/node_endpoint.go

```diff
@@ -443,6 +443,11 @@ func (n *Node) UpdateDrain(args *structs.NodeUpdateDrainRequest,
 		}
 	}
 
+	// Mark the deadline time
+	if args.DrainStrategy != nil && args.DrainStrategy.Deadline.Nanoseconds() > 0 {
```

Comment:
Hm, I've been using `IsZero()` instead of the Nanoseconds comparison... hopefully they're equivalent?